scripts/install.sh

```diff
@@ -0,0 +1,183 @@
+
+Usage:
+  $0 [OPTIONS]
+
+Options:
+  --cli <oc|kubectl>          : Specify the CLI tool to use (overrides auto-detection).
+  --router-base <router-base> : Manually provide the cluster router base if auto-detection fails.
+  --release-name <name>       : Specify a custom release name for the Helm chart.
+  --generate-name             : Generate a name for the Helm release (overrides --release-name).
+  --namespace <namespace>     : Specify the namespace for the Helm release (autodetects if not provided).
+  --help                      : Show this help message and exit.
+
+Examples:
+  $0                               # Auto-detects router base and installs the Helm chart
+  $0 --cli kubectl                 # Uses kubectl even if oc is installed
+  $0 --router-base example.com     # Manually specifies the router base and installs the Helm chart
+  $0 --release-name myrelease      # Installs the Helm chart with the specified release name
+  $0 --generate-name               # Generates a name for the Helm release
+"
+}
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check if required files and directories exist
+HELM_CHART_DIR="../charts/backstage"
+VALUES_FILE="$HELM_CHART_DIR/values.yaml"
+
+if [ ! -d "$HELM_CHART_DIR" ]; then
+    echo "Error: Helm chart directory not found at $HELM_CHART_DIR"
+    exit 1
+fi
+
+if [ ! -f "$VALUES_FILE" ]; then
+    echo "Error: values.yaml file not found at $VALUES_FILE"
+    exit 1
+fi
+
+# Default CLI detection
+detect_cli() {
+    if command_exists oc; then
+        CLI="oc"
+    elif command_exists kubectl; then
+        CLI="kubectl"
+    else
+        echo "Error: Neither 'oc' nor 'kubectl' is installed. Please install one of them to proceed."
+        exit 1
+    fi
+}
+
+# Parse command-line arguments
+CLI=""
+ROUTER_BASE=""
+RELEASE_NAME=""
+GENERATE_NAME=false
+NAMESPACE=""
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --cli)
+            CLI="$2"
+            if [[ "$CLI" != "oc" && "$CLI" != "kubectl" ]]; then
+                echo "Error: Invalid value for --cli. Use 'oc' or 'kubectl'."
+                exit 1
+            fi
+            shift
+            ;;
+        --router-base)
+            ROUTER_BASE="$2"
+            shift
+            ;;
+        --release-name)
+            RELEASE_NAME="$2"
+            shift
+            ;;
+        --generate-name)
+            GENERATE_NAME=true
+            ;;
+        --namespace)
+            NAMESPACE="$2"
+            shift
+            ;;
+        --help)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "Unknown parameter passed: $1"
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+# If CLI is not specified, detect automatically
+if [[ -z "$CLI" ]]; then
+    detect_cli
+fi
+
+# Check if the user is logged into a cluster
+if ! $CLI whoami >/dev/null 2>&1; then
```

Comment:
You are correct - Since the goal of this script is to ensure that users are prepared before they proceed with the installation, checking if the user is logged in might not be necessary because Helm and other commands will fail with appropriate error messages. I will remove this logic as it is not necessary.